src/interfaces/UserInterface.ts

```diff
@@ -1,11 +1,20 @@
+import { Request } from 'express'
 import { SlideInterface } from './SlideInterface'
 
 export interface UserInterface {
-  id: string
+  id?: string
   name: string
   email: string
   password: string
-  slides: SlideInterface[]
+  slides?: SlideInterface[]
+}
+export interface UserOptions {
+  id?: number
+  email?: string
+  name?: string
+}
+export interface UserRequest extends Request {
+  body: UserInterface
```

Comment:
As the `id` is auto generated, and you don't need it in the body request, you can use the Omit function to take it out from the UserInterface and still use the same interface with the request body.
``` Omit< UserInterface,  'id'>```